OpenRA.Mods.Common/Traits/HitShape.cs

```diff
@@ -129,11 +132,24 @@ public WDist DistanceFromEdge(Actor self, WPos pos)
 			return Info.Type.DistanceFromEdge(pos, origin, orientation);
 		}
 
+		public IEnumerable<IRenderable> RenderDebugAnnotations(Actor self, WorldRenderer wr)
+		{
+			var targetPosHLine = new WVec(0, 128, 0);
+			var targetPosVLine = new WVec(128, 0, 0);
+			var targetPosColor = IsTraitDisabled ? Color.Gainsboro : Color.Lime;
+			var targetablePositions = TargetablePositions(self).ToArray();
```

Comment:
Is there a reason you `ToArray` this? The enumerable is only used once, isn't it?

Reply:
I had trouble to get the foreach to work in a previous revision, but it looks like that was caused by something else.
Updated.